### PR TITLE
chore: Centralize Status Management

### PR DESCRIPTION
**Pull request by gatici**
# Description

This PR uses new ops feature CollectStatus Event to manage status Charm. Event named collect_unit_status sets the status of charm automatically at the end of every hook.

Reference:
   -  https://ops.readthedocs.io/en/latest/#ops.CollectStatusEvent
   -  https://discourse.charmhub.io/t/how-to-set-a-charms-status/11771

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: tests/integration/test_integration.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ async def test_when_scale_app_beyond_1_then_only_one_unit_is_active(
     )
     unit_statuses = Counter(unit.workload_status for unit in app.units)
     assert unit_statuses.get("active") == 1
-    assert unit_statuses.get("blocked") == 2
+    assert unit_statuses.get("unknown") == 2
```
Comment (Gmerold) [inline tests/integration/test_integration.py:178]: Hmmm, that's not good, isn't it? The status should actually be `Blocked` in this case. 
In the charm, if we return from the constructor right away, is `CollectStatusEvent` fired?

**Review by Gmerold — CHANGES_REQUESTED** [review]

File: tests/integration/test_integration.py (hunk 1 above)
Comment (gatici) [reply tests/integration/test_integration.py:178]: According to documentation https://ops.readthedocs.io/en/latest/#ops.CollectStatusEvent  collect_app_status will only be triggered on the leader unit. That explains the reason.
Now the question is how could we set a status for non-leader units ?

Comment (Gmerold) [reply tests/integration/test_integration.py:178]: In the constructor - just as we used to do :)

Comment (gatici) [reply tests/integration/test_integration.py]: Found the solution. Registering collect_status observer before we return in the constructor should fix the issue.

**Review by Gmerold — APPROVED** [review]

**Pull request merged**